lib/ex_machina/ecto.ex

```diff
@@ -69,10 +69,13 @@ defmodule ExMachina.Ecto do
   defp drop_ecto_fields(record = %{__struct__: struct, __meta__: %{__struct__: Ecto.Schema.Metadata}}) do
     record
     |> Map.from_struct
-    |> Map.delete(:__meta__)
-    |> Map.drop(struct.__schema__(:associations))
+    |> Map.drop(ecto_fields(struct))
   end
   defp drop_ecto_fields(record) do
     raise ArgumentError, "#{inspect record} is not an Ecto model. Use `build` instead."
   end
+
+  defp ecto_fields(struct) do
+    [:__meta__] ++ struct.__schema__(:associations) ++ struct.__schema__(:primary_key)
+  end
```

Comment:
I think this feels a bit roundabout because we already knowing it's dropping ecto fields because this function is called from `drop_ecto_fields`. So I think the abstraction is probably not necessary. What do you think about

```elixir
record
|> Map.from_struct
|> Map.delete(:__meta__)
|> Map.delete(struct.schema__(:associations))
|> Map.delete(struct.schema__(:primary_key))
```

However, I don't feel strongly so if you do, feel free to leave it as is